base/sparse/sparsematrix.jl

```diff
@@ -3507,7 +3519,7 @@ end
 
 function trace{Tv}(A::SparseMatrixCSC{Tv})
     if size(A,1) != size(A,2)
-        throw(DimensionMismatch("expected square matrix"))
+        throw(DimensionMismatch("matrix is not square"))
```

Comment:
I guess we don't allow this for dense right now either, but isn't trace well-defined even for rectangular matrices?